plugins/org.jkiss.dbeaver.core/src/org/jkiss/dbeaver/ui/dialogs/connection/ConnectionPageSettings.java

```diff
@@ -192,6 +191,31 @@ public void activatePage() {
         //getContainer().updateTitleBar();
     }
 
+    private void createAdditionalRequiredPages() {
+        if (connectionEditor instanceof IDialogPageProvider dialogPageProvider) {
+            IDialogPage[] requiredDialogPages = dialogPageProvider.getRequiredDialogPages();
+            // potentially n^2 but we don't have a lot of required pages...
+            if (requiredDialogPages != null) {
+                for (IDialogPage requiredDialogPage : requiredDialogPages) {
+                    for (CTabItem item : tabFolder.getItems()) {
+                        if (item.getData().equals(requiredDialogPage)) {
+                            Composite panel = (Composite) item.getControl();
+                            try {
+                                panel.setRedraw(false);
+                                requiredDialogPage.createControl(panel);
+                                Dialog.applyDialogFont(panel);
+                                panel.layout(true, true);
+                                requiredDialogPage.setVisible(false);
+                            } finally {
+                                panel.setRedraw(true);
```

Comment:
The finally - clause generally defined for completing exceptional operation like: close resource, streams etc Are you sure for setRedraw here?

Reply:
I think it is ok. We use it widely.
finally is not for resources close but for any operations which must be called in the end of block.